doc/ss_logger.rst

```diff
@@ -0,0 +1,201 @@
+logging and is intended for use during development.
+
+Next, set `SMARTSIM_LOG_LEVEL` to `quiet` in terminal::
+    export SMARTSIM_LOG_LEVEL=quiet
+
+When we run the program once again, the following output is printed
+to stdout::
+    11:15:00 system.host.com SmartSim[65385] ERROR This is an error message
+    11:15:00 system.host.com SmartSim[65385] WARNING This is a warning message
+
+Notice that the `info` and `debug` log functions were filtered. This is because by using
+the highest severity level (`quiet`), we instruct SmartSim to omit the lower severity levels
+(`info`, `debug` and `developer`).
+
+To finish the example, set `SMARTSIM_LOG_LEVEL` to `info` in terminal::
+    export SMARTSIM_LOG_LEVEL=info
+
+When we execute the script, the following messages will print
+to stdout::
+    11:15:00 system.host.com SmartSim[130033] INFO This is a message
+    11:15:00 system.host.com SmartSim[130033] ERROR This is an error message
+    11:15:00 system.host.com SmartSim[130033] WARNING This is a warning message
+
+Notice that the same messages were logged to stdout as when we ran the script with the default value `info`.
+SmartSim omits the lower severity levels (`debug` and `developer`).
+
+---------------
+Logging to File
+---------------
+.. _log_to_file:
+The ``log_to_file()`` function in SmartSim allows users to log messages
+to a specified file by providing a file name or relative file path. If the file name
+passed in does not exist, SmartSim will create the file. If the program is re-executed with the same
+file name, the file contents will be overwritten.
+
+To demonstrate, begin by importing the functions `get_logger` and `log_to_file` from the `log` module:
+
+.. code-block:: python
+
+      from smartsim.log import get_logger, log_to_file
+
+Initialize a logger for use within the Python driver script:
+
+.. code-block:: python
+
+      logger = get_logger("SmartSim")
+
+Add the ``log_to_file()`` function to instruct SmartSim to create a file named `logger.out`
+to write log messages to:
+
+.. code-block:: python
+
+      log_to_file("logger.out")
+
+For the example, we add all log functions to the script:
+
+.. code-block:: python
+
+      logger.info("This is a message")
+      logger.debug("This is a debug message")
+      logger.error("This is an error message")
+      logger.warning("This is a warning message")
+
+Remember that the default value for the `SMARTSIM_LOG_LEVEL` variable is `info`.
+Therefore, we will not set the environment variable and instead rely on the
+default.
+
+When we execute the Python script, a file named `logger.out` is created in our working
+directory with the listed contents::
+    11:15:00 system.host.com SmartSim[10950] INFO This is a message
+    11:15:00 system.host.com SmartSim[10950] ERROR This is an error message
+    11:15:00 system.host.com SmartSim[10950] WARNING This is a warning message
+
+Notice that the `debug` function message was filtered. This is because by using
```

Comment:
Notice that the `debug` function message was filtered. This is because by using
a higher severity level (`info`), we instruct SmartSim to omit the lower severity levels (`debug` and `developer`).

->

Notice that the `debug` function message was filtered. This is because by using
a lower verbosity level (`info`), we instruct SmartSim to omit the higher verbosity messages (`debug` and `developer`).